python/lance/data/convert/nuscenes.py

```diff
@@ -0,0 +1,267 @@
+                sample_joined = self._merge_dict(sample_joined, sensor, "sensor")
+
+                # Find the annotations for the sample
+                object_anns = self._find_annotations(sample_data["token"], "sample_data_token", "object_ann")
+                surface_ann = self._find_annotations(sample_data["token"] , "sample_data_token", "surface_ann")
+                sample_joined["object_ann"] = object_anns
+                sample_joined["surface_ann"] = surface_ann
+                
+                instances_df.append(sample_joined)
+        
+        return pd.DataFrame(instances_df)
+    
+    def get_schema(self) -> pa.Schema:
+        """
+        Returns the Arrow schema.
+        This is flatten/denormalized for better ergonomics for
+        ML - i.e. filtering/slicing the dataset and reducing joins.
+        Column names are renamed accordingly to prevent collisions.
+        """
+        # Mask
+        mask_schema = pa.struct([
+            pa.field("size", pa.list_(pa.int32(), 2)),
+            pa.field("counts", pa.string())
+        ])
+
+        # Surface Ann
+        surface_ann_schema = pa.struct([
+            pa.field("token", pa.string()),
+            pa.field("sample_data_token", pa.string()),
+            pa.field("category_token", pa.string()),
+            pa.field("mask", mask_schema), # TODO: change to RLE once type is created 
+
+            # Category
+            pa.field("category_token_", pa.string()),
+            pa.field("category_name_", pa.string()),
+            pa.field("category_description_", pa.string())
+        ])
+
+        # Object Ann
+        object_ann_schema = pa.struct([
+            pa.field("token", pa.string()),
+            pa.field("sample_data_token", pa.string()),
+            pa.field("category_token", pa.string()),
+            pa.field("attribute_tokens", pa.list_(pa.string())),
+            pa.field("bbox", Box2dType()),
+            pa.field("mask", mask_schema), # TODO: change to RLE once type is created
+
+            # Category
+            pa.field("category_token_", pa.string()),
+            pa.field("category_name_", pa.string()),
+            pa.field("category_description_", pa.string())
+        ])
+
+        return pa.schema([
+            # Sample is our root table, we don't rename these fields
+            pa.field("token", pa.string()),
+            pa.field("timestamp", pa.int64()),
+            pa.field("log_token", pa.string()),
+            pa.field("key_camera_token", pa.string()),
+            
+            # Log
+            pa.field("log_token_", pa.string()),
+            pa.field("log_logfile_", pa.string()),
+            pa.field("log_vehicle_", pa.string()),
+            pa.field("log_date_captured_", pa.string()),
+            pa.field("log_location_", pa.string()),
+            
+            # Calibrated Sensor
+            pa.field("calibrated_sensor_token_", pa.string()),
```

Comment:
Are these token used for joins? after a flatten table, should we still keep them in the dataset?

Reply:
yep, they are. removed now